agent_ops.go

```diff
@@ -20,6 +21,16 @@ func (agent *Agent) GetReplica(ctx context.Context, opts *GetReplicaOptions) (*G
 	return agent.crud.GetReplica(ctx, opts)
 }
 
+func (agent *Agent) GetAllReplicas(ctx context.Context, opts *GetAllReplicasOptions) ([]GetAllReplicasResult, error) {
+	bucket, err := agent.mgmt.GetBucket(ctx, &cbmgmtx.GetBucketOptions{
```

Comment:
This should not be neccessary.  If you take a look at how CrudComponent executes a GetReplica operation, it's execution stack eventually invokes OrchestrateMemdRouting, which is responsible for identifying the place that a particular operation is being sent.  Additionally, this implementation would still have the same problem mentioned in the ING where a change to the servers for a replica could cause GetAllReplicas to miss a replica or duplicate the result from a specific replica.  What we should instead do is use the same topological information that OrchestrateMemdRouting uses to identify which nodes to send an operation to in a 'live' manner, and additionally the logic should be adjusted such that we do not refer to the replicas we are trying to fetch by their arbitrary 'replicaIdx' but rather by the server that the request is being sent to.  We already deal with this a bit in our handling of routing, where we route operations to a nodeId rather than to a specific vbucket number, such that we can maintain the 'atomicity' of this information.

Reply:
Yeah I thought I was missing the last thing. Have made some changes